online-shop/src/app/auth/models/User.model.ts

```diff
@@ -0,0 +1,5 @@
+export interface User {
+    username:string
+    fullName:string
+    roles:(String)[]
```

Comment:
Replace this 'String' wrapper object with primitive type 'string'.
it is recommended to use lowercase string to tell Typescript you want to get an actual string. If you use String, then you will get an object.
`let name = new String('Mike');
console.log(typeof name); // object (not a string)`
*for all places (if needed)